x-pack/plugin/rollup/src/main/java/org/elasticsearch/xpack/rollup/action/TransportRollupSearchAction.java

```diff
@@ -454,6 +454,9 @@ public void onResponse(SearchResponse response) {
                         channel.sendResponse(response);
                     } catch (Exception e) {
                         onFailure(e);
+                    } finally {
+                        // TODO - avoid the implicit incref elsewhere and then replace this whole thing with a ChannelActionListener
```

Comment:
Is the counterpart not the explicit incref here:

https://github.com/elastic/elasticsearch/blob/1d7d9dc1eed2aeda08dd7a62a8a1893c5113a95b/x-pack/plugin/rollup/src/main/java/org/elasticsearch/xpack/rollup/action/TransportRollupSearchAction.java#L162

I wonder if removing that (and fixing the todo) would work, but I imagine you having tried it out so curious about the result/where the implicit incref is?

Reply:
I think this one is also germane:

https://github.com/elastic/elasticsearch/blob/1d7d9dc1eed2aeda08dd7a62a8a1893c5113a95b/x-pack/plugin/rollup/src/main/java/org/elasticsearch/xpack/rollup/RollupResponseTranslator.java#L245

But this code is also kinda branchy and I didn't want to spend too much time disentangling how the ownership of these refs is handed around. I opened https://github.com/elastic/elasticsearch/issues/104503 to track this debt.

Reply:
NB I also don't trust the tests in this area, I tried making some obviously incorrect changes to this refcounting and everything still passed.

Reply:
Sounds good, I also tried fiddling with the incref and it did not fail (which it sounds like it should have). But happy to leave it to the search team to fix that.